gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiServiceImpl.java

```diff
@@ -2004,75 +2020,85 @@ public void delete(ExecutionContext executionContext, String apiId) {
 
             Api api = apiRepository.findById(apiId).orElseThrow(() -> new ApiNotFoundException(apiId));
 
-            if (api.getLifecycleState() == LifecycleState.STARTED) {
-                throw new ApiRunningStateException(apiId);
+            if (DefinitionContext.isKubernetes(api.getOrigin())) {
+                // Quick win: when api is managed by Kubernetes, for now, just mark the api STOPPED, remove it from the search engine and mark it UNPUBLISHED from dev portal.
+                // This must be better handled with an higher concept such as 'archiving' or something close.
+                api.setLifecycleState(LifecycleState.STOPPED);
+                api.setApiLifecycleState(ApiLifecycleState.UNPUBLISHED);
+                api.setVisibility(Visibility.PRIVATE);
+                apiRepository.update(api);
```

Comment:
Plan seems not removed. this part was manage by this issue : https://github.com/gravitee-io/issues/issues/8139